src/main/java/com/scaler/lld/design/assignments/singleton/FileBasedConfigurationManagerImpl.java

```diff
@@ -5,46 +5,69 @@ public class FileBasedConfigurationManagerImpl extends FileBasedConfigurationMan
     @Override
     public String getConfiguration(String key) {
         // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+        return properties.getProperty(key);
     }
 
     @Override
     public <T> T getConfiguration(String key, Class<T> type) {
         // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+        if(INSTANCE!=null){
+            String str=INSTANCE.properties.getProperty(key);
```

Comment:
Instead of calling `INSTANCE.properties.getProperty(key);`, you can just call `this.getProperty(key);` since `getConfiguration` is an instance method and not static.